frontend/packages/local-storage-operator-plugin/src/components/disks-list/disks-list-page.tsx

```diff
@@ -123,26 +128,41 @@ export const NodesDisksListPage: React.FC<NodesDisksListPageProps> = ({
   const nodeRole = getNodeRole(obj);
   const propName = `lvdr-${nodeName}`;
 
+  const makeLocalVolumeDiscoverRequest = async (ns: string) => {
+    const nodeNameByHostnameLabel = obj.metadata?.labels?.['kubernetes.io/hostname'];
+    setProgress(true);
+    try {
+      await updateLocalVolumeDiscovery([nodeNameByHostnameLabel], ns, setError);
+    } catch (error) {
+      if (error?.response?.status === 404) {
+        try {
+          await createLocalVolumeDiscovery([nodeNameByHostnameLabel], ns, setError);
+        } catch (createError) {
+          setError(createError.message);
+          setProgress(false);
+        }
+      } else {
+        setError(error.message);
+        setProgress(false);
+      }
```

Comment:
Something that can be taken into consideration for future refactor. I really don't like the cascaded catch and try blocks it would be better if we had a single try and catch block.

Reply:
Can you suggest something , so we can take it another PR  and fix all other places too?
Right now we are using this at a couple of places because we need to look out for the availability of CR hence cascading but it helps with readability and debugging.

Thanks

Reply:
I refactored the syntax a bit, but the logic is the same. Is this what you had in mind?

```typescript
  const makeLocalVolumeDiscoverRequest = async (ns: string) => {
    const nodeNameByHostnameLabel = obj.metadata?.labels?.['kubernetes.io/hostname'];
    const reduceTryCatch = async (
      tryer: { (): Promise<void>; (): Promise<void>; (): any },
      catcher: { (error: any): void; (createError: any): void; (arg0: any): void },
    ) => {
      try {
        await tryer();
      } catch (err) {
        catcher(err);
      }
    };
    setProgress(true);
    reduceTryCatch(
      async () => {
        await updateLocalVolumeDiscovery([nodeNameByHostnameLabel], ns, setError);
      },
      (error: { response: { status: number }; message: React.SetStateAction<string> }) => {
        if (error?.response?.status === 404) {
          reduceTryCatch(
            async () => {
              await createLocalVolumeDiscovery([nodeNameByHostnameLabel], ns, setError);
            },
            (createError: { message: React.SetStateAction<string> }) => {
              setError(createError.message);
              setProgress(false);
            },
          );
        } else {
          setError(error.message);
          setProgress(false);
        }
      },
    );
  };
```